### PR TITLE
Fix config.hasOwnProperty is not a function

### LINKED ISSUE #6: config.hasOwnProperty is not a function
Using the lib with an option object containing a valid path throws the error 

> TypeError: config.hasOwnProperty is not a function

```
var getEmail = require("git-user-email");
var gitConfigPath = require("git-config-path");

var email = getEmail({ path: gitConfigPath("global") });
```

### PR DESCRIPTION
**Pull request by ThomasKientz**
This project has a dependency over [parse-git-config's package](https://github.com/jonschlinkert/parse-git-config) which has a dependency over [ini's package](https://github.com/npm/ini) which has been updated yesterday and break the use of `hasOwnProperty()`

https://github.com/npm/ini/commit/032fbaf5f0b98fce70c8cc380e0d05177a9c9073#diff-508d78eecdeb317e6a815a221bd6927f51e8780e4eb54dcdc6f30990c442fdc8R70

Fixes https://github.com/jonschlinkert/git-user-email/issues/6

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ var extend = require('extend-shallow');
 module.exports = function gitUserEmail(opts) {
   opts = extend({cwd: '/', path: gitconfig()}, opts);
   var config = parse.sync(opts);
-  if (typeof config === 'object' && config.hasOwnProperty('user')) {
+  if (typeof config === 'object' && Object.hasOwnProperty.bind(config)("user")) {
```
Comment (doowb) [inline index.js]: I think using `.call` is fine here.

```suggestion
  if (typeof config === 'object' && Object.hasOwnProperty.call(config, 'user')) {
```

**Comment by doowb** [comment]
Hi @ThomasKientz and thanks for the PR!

I think this is a good change to make (with my small suggestion above).

However, I also believe that the change in the `ini` package should have been a major version bump since it's a breaking change in what their api returns. Since I'm sure this will cause issues in more packages, will you open an issue on `ini` suggesting a rollback and publishing that change as a major version bump? (I also noticed that his tests are failing in that version).

Thanks.

**Comment by gr2m** [comment]
done via  https://github.com/npm/ini/issues/103#issuecomment-743397410

> Fixed on 1.3.8, I'll publish a v2 shortly that re-introduces the change. (For what it's worth, relying on `obj.hasOwnProperty` is a bit brittle anyway, because someone could put a `hasOwnProperty` field in the ini file itself.)j

**Comment by jonschlinkert** [comment]
Going to go ahead and merge, since I appreciate the contribution and I'm going to do some other updates to this lib anyway. thank you

**Pull request merged**